### ISSUE TITLE
[BUG] salt-ssh 3005.4 does not work when executed from RHEL 9.2 machine

### ISSUE THREAD

**Issue by golmaal** [saltstack/salt#65787]
**Description**
We are unable to execute the salt-ssh commands when executing salt-ssh from RHEL 9.2 machine with python3.9

salt-ssh 10.2.0.53 --user root --passwd password --ignore-host-keys test.ping -ltrace

Expecting the above to output True.

The 10.2.0.53 has python3.9 installed. 
The same command succeeds if run from CentOS 8.5 based salt-ssh 3004.2. 

**Setup**
Install RHEL 9.2 minimal in a KVM VM
Install salt 3005.4 from epel-9 repo

Setup another RHEL 9.2 VM. In this case it's 10.2.0.53.

**Steps to Reproduce the behavior**
Execute the following command on salt master - 

salt-ssh 10.2.0.53 --user root --passwd password --ignore-host-keys test.ping -ltrace

Please see the attached file for trace of the command 
[output.txt](https://github.com/saltstack/salt/files/13823802/output.txt)

**Expected behavior**
Expecting to see True printed as the output of test.ping command.

**Versions Report**
<details><summary>salt --versions-report</summary>
(Provided by running salt --versions-report. Please also mention any differences in master/minion versions.)

```yaml
salt --versions-report
Salt Version:
          Salt: 3005.4

Dependency Versions:
          cffi: 1.16.0
      cherrypy: unknown
      dateutil: 2.8.1
     docker-py: Not Installed
         gitdb: Not Installed
     gitpython: Not Installed
        Jinja2: 2.11.3
       libgit2: Not Installed
      M2Crypto: 0.38.0
          Mako: 1.1.4
       msgpack: 1.0.3
  msgpack-pure: Not Installed
  mysql-python: Not Installed
     pycparser: 2.21
      pycrypto: Not Installed
  pycryptodome: 3.14.0
        pygit2: Not Installed
        Python: 3.9.16 (main, Sep 12 2023, 00:00:00)
  python-gnupg: 0.5.2
        PyYAML: 5.4.1
         PyZMQ: 22.3.0
         smmap: Not Installed
       timelib: Not Installed
       Tornado: 4.5.3
           ZMQ: 4.3.4

System Versions:
          dist: rhel 9.2 Plow
        locale: utf-8
       machine: x86_64
       release: 5.14.0-284.30.1.el9_2.x86_64
        system: Linux
       version: Red Hat Enterprise Linux 9.2 Plow
```
</details>

**Additional context**
We're in the process of migrating from CentOS 8.5 to RHEL 9.x.
This issue is blocking our progress.

**Comment by golmaal** [comment]
@Ch3LL - this issue maybe a duplicate of https://github.com/saltstack/salt/issues/65406
Using a complex password, I'm unblocked.

**Comment by dwoz** [comment]
@golmaal Is this also an issue in 3006.5? The 3005 release is CVE only support at this point.

**Comment by dmurphy18** [comment]
@golmaal Salt 3005 is now EOL, can you retry with Salt 3006.7
